daphne_worker/src/dap.rs

```diff
@@ -195,19 +204,31 @@ impl<D> DapAggregator<BearerToken> for DaphneWorkerConfig<D> {
         let agg_shares =
             DapAggregateShare::batches_from_out_shares(out_shares, task_config.min_batch_duration)?;
 
-        for (window, agg_share) in agg_shares.into_iter() {
-            let stub = namespace
-                .id_from_name(&durable_agg_store_name(
-                    &task_config.version,
-                    &task_id.to_hex(),
-                    window,
-                ))?
-                .get_stub()?;
-
-            // TODO Don't block on DO requests (issue multiple requests simultaneously).
-            durable_post!(stub, DURABLE_AGGREGATE_STORE_MERGE, &agg_share).await?;
+        let mut stubs_and_shares = Vec::with_capacity(agg_shares.len());
+        for (window, agg_share) in agg_shares.clone().into_iter() {
+            stubs_and_shares.push((
+                namespace
+                    .id_from_name(&durable_agg_store_name(
+                        &task_config.version,
+                        &task_id.to_hex(),
+                        window,
+                    ))?
+                    .get_stub()?,
+                agg_share,
+            ));
+        }
+
+        let mut requests = Vec::with_capacity(stubs_and_shares.len());
+        for stub_and_share in &stubs_and_shares {
+            requests.push(durable_post!(
+                (*stub_and_share).0, // Stub
+                DURABLE_AGGREGATE_STORE_MERGE,
+                &(*stub_and_share).1 // AggregateShare
+            ));
         }
 
+        let _responses = try_join_all(requests).await?;
```

Comment:
Let's make sure each response status is 200 and abort with internal error if not.

Reply:
Fixed.

Reply:
I'm sorry, I appear to have mislead you here!! There is actually nothing to validate here: If the response was not 200 then parsing the JSON blob from the response would fail and we'd err here.

My bad. What this comment should have been is: There's nothing to validate here, so you can remove the `let` statement.

Reply:
Thanks for clarifying! Please see my response below.

Reply:
This should be reflected in the latest commit.